external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -2601,36 +2587,38 @@ fn parse_dot_or_index_chain(context: &mut Context) -> Result<Exp, Box<Diagnostic
                             }
                         }
                     }
-                    _ => {
-                        let n = parse_identifier(context)?;
-                        if is_start_of_call_after_function_name(context, &n) {
-                            let call_start = context.tokens.start_loc();
-                            let is_macro = if let Tok::Exclaim = context.tokens.peek() {
-                                let loc = current_token_loc(context.tokens);
-                                context.tokens.advance()?;
-                                Some(loc)
+                    _ => match parse_identifier(context) {
+                        Err(diag) => {
+                            context.add_diag(*diag);
+                            Exp_::DotUnresolved(loc, Box::new(lhs))
+                        }
+                        Ok(n) => {
+                            if is_start_of_call_after_function_name(context, &n) {
+                                let call_start = context.tokens.start_loc();
+                                let is_macro = if let Tok::Exclaim = context.tokens.peek() {
+                                    let loc = current_token_loc(context.tokens);
+                                    context.advance();
+                                    Some(loc)
+                                } else {
+                                    None
+                                };
+                                let mut tys = None;
+                                if context.tokens.peek() == Tok::Less
+                                    && n.loc.end() as usize == call_start
+                                {
+                                    tys = parse_optional_type_args(context);
+                                }
```

Comment:
Is there a reason not to just call `parse_macro_opt_and_tyargs_opt` here?

Reply:
Not that I can see. Previously, I just used what was already there, but now I rewrote it to use `parse_macro_opt_and_tyargs_opt`.